src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -1004,14 +1003,84 @@ public IEnumerable<Index> GetIndexes()
 
         public void RunIdleOperations()
         {
-            AsyncHelpers.RunSync(HandleUnusedAutoIndexes);
+            DatabaseRecord record;
+            using (_serverStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
+                record = _serverStore.Cluster.ReadDatabase(context, _documentDatabase.Name);
+
+            if (record.Topology.Members[0] != _serverStore.NodeTag)
+                return;
+
+            AsyncHelpers.RunSync(RunIdleOperationsAsync);
         }
 
-        private async Task HandleUnusedAutoIndexes()
+        private async Task RunIdleOperationsAsync()
```

Comment:
Technically speaking, this may actually run on two different machines at the same time, even with the check.
I wonder if we should handle this in some manner? Maybe do the operation optimistically, so send it to the leader and it will only actually run if the db record didn't change?

Reply:
I have some ideas, let's apply them as a part of http://issues.hibernatingrhinos.com/issue/RavenDB-11206